repositories/productRepository.js

```diff
@@ -0,0 +1,23 @@
+let products = [];
+
+export const getProducts = () => {
+    return products;
+};
+
+export const getProductById = (id) => {
+    return products.find(product => product.id === id);
+};
+
+export const addProduct = (product) => {
+    products.push(product);
+};
+
+export const updateProduct = (id, updatedProduct) => {
+    products = products.map(product => 
```

Comment:
I would suggest to use find() instead of map() because you need to update only one product, so if it is the first element in the array, there is no reason to iterate  over remaining items